tensorflow/core/kernels/data/batch_dataset_op_test_v2.cc

```diff
@@ -57,126 +57,44 @@ class BatchDatasetParams : public DatasetParams {
 
 class BatchDatasetOpTest : public DatasetOpsTestBase {
  public:
-  Status SetupTestEnv(int thread_num, int cpu_num,
-                      const std::vector<FunctionDef>& flib) {
-    TF_RETURN_IF_ERROR(InitThreadPool(thread_num));
-    TF_RETURN_IF_ERROR(InitFunctionLibraryRuntime(flib, cpu_num));
-  }
+  Status Initialize(DatasetParams* dataset_params) override {
+    auto batch_dataset_params =
+        dynamic_cast<BatchDatasetParams*>(dataset_params);
```

Comment:
Maybe we can avoid `dynamic_cast` by changing `DatasetOpsTestBase` to be a template class as below:
```C++
templcate <typename T>
class DatasetOpsTestBase : public ::testing::Test { 
  ...
  virtual Status Initialize(T* params) = 0;
 ...
}
```
Then `BatchDatasetOpTest` can be changed to 
```C++
class BatchDatasetOpTest : public DatasetOpsTestBase<BatchDatasetParams> {
  ...
  Status Initialize(BatchDatasetParams* dataset_params) override {
   ...
  }
}
```
However, this change will break other existing tests. We could do it when all the existing tests are refactored to have the `Initialize ()` function.

Reply:
How about creating DatasetOpsTestBaseV2 that is templatized and incrementally migrate all tests to it?

Reply:
Sounds good! Will do it.